airflow/www/static/js/dag/details/SetDagTaskNotes.tsx

```diff
@@ -0,0 +1,134 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionPanel,
+  AccordionIcon,
+  Box,
+  Button,
+  Text,
+  Textarea,
+} from '@chakra-ui/react';
+import ResizeTextarea from 'react-textarea-autosize';
+import { getMetaValue } from '../../utils';
+import { useSetDagRunNotes, useSetTaskInstanceNotes } from '../../api';
+
+interface Props {
+  dagId: string;
+  runId: string;
+  taskId: string | undefined;
+  mapIndex: number | undefined;
+  initialValue: string | undefined | null;
+}
+
+const canEdit = getMetaValue('can_edit') === 'True';
+
+const SetDagTaskNotes = ({
+  dagId, runId, taskId, mapIndex, initialValue,
+}: Props) => {
+  const [notes, setNotes] = useState(initialValue == null ? '' : initialValue);
+  const [noteBeforeEdit, setNoteBeforeEdit] = useState('');
+  const [editMode, changeEditMode] = useState(false);
+  const tiStr = (taskId != null ? taskId : '');
+  const {
+    mutateAsync: apiCallToSetDagRunNote, isLoading: dagRunIsLoading,
+  } = useSetDagRunNotes(dagId, runId, notes);
+  const {
+    mutateAsync: apiCallToSetTINote, isLoading: tiIsLoading,
+  } = useSetTaskInstanceNotes(dagId, runId, tiStr, (mapIndex != null ? mapIndex : -1), notes);
+
+  const objectIdentifier = (taskId == null) ? 'DAG Run' : 'Task Instance';
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (taskId == null) {
+      await apiCallToSetDagRunNote();
+    } else {
+      await apiCallToSetTINote();
+    }
+    changeEditMode(false);
+  };
+
+  return (
+    <Accordion defaultIndex={[0]} allowMultiple>
+      <AccordionItem style={{ border: 0 }}>
+        <AccordionButton style={{ padding: 0, paddingBottom: '10px', fontSize: 'inherit' }}>
+          <Box flex="1" textAlign="left">
+            <Text as="strong" size="lg">
+              {objectIdentifier}
+              {' '}
+              Notes:
+            </Text>
+          </Box>
+          <AccordionIcon />
+        </AccordionButton>
+        <AccordionPanel style={{ padding: '0 0 15px 0' }}>
+          {editMode ? (
+            <form onSubmit={handleSubmit}>
+              <div>
+                <Textarea
+                  minH="unset"
+                  overflow="hidden"
+                  w="100%"
+                  resize="none"
+                  minRows={3}
+                  maxRows={10}
+                  as={ResizeTextarea}
+                  value={notes}
+                  onChange={(e) => setNotes(e.target.value)}
+                  style={{ width: '100%' }}
+                />
+              </div>
+              <div style={{ marginTop: '10px' }}>
+                <Button type="submit" isLoading={dagRunIsLoading || tiIsLoading}>
+                  Update user notes
```

Comment:
Looks like almost all button in UI are in Title Case -> `Update User Notes`. Maybe we should update these new buttons to harmonize with the rest of the UI ?

Reply:
Yes indeed. Changed